packages/bsky/src/db/coordinator.ts

```diff
@@ -0,0 +1,80 @@
+import { Migrator } from 'kysely'
+import PrimaryDatabase from './primary'
+import Database from './db'
+import { PgOptions } from './types'
+
+type ReplicaTag = 'timeline'
+type ReplicaOptions = PgOptions & { tag?: ReplicaTag }
+
+type CoordinatorOptions = {
+  schema?: string
+  primary: PgOptions
+  replicas: ReplicaOptions[]
+}
+
+type ReplicaGroup = {
+  dbs: Database[]
+  roundRobinIdx: number
+}
+
+export class DatabaseCoordinator {
+  migrator: Migrator
+  destroyed = false
+
+  private primary: PrimaryDatabase
+  private allReplicas: Database[]
+  private tagged: Record<string, ReplicaGroup>
+  private untagged: ReplicaGroup
+
+  constructor(public opts: CoordinatorOptions) {
+    this.primary = new PrimaryDatabase({
+      schema: opts.schema,
+      ...opts.primary,
+    })
+    this.allReplicas = []
+    this.tagged = {}
+    this.untagged = {
+      dbs: [],
+      roundRobinIdx: 0,
+    }
+    for (const cfg of opts.replicas) {
+      const db = new Database({
+        schema: opts.schema,
+        ...cfg,
+      })
+      this.allReplicas.push(db)
+      if (cfg.tag) {
+        this.tagged[cfg.tag] ??= {
+          dbs: [],
+          roundRobinIdx: 0,
+        }
+        this.tagged[cfg.tag].dbs.push(db)
+      } else {
+        this.untagged.dbs.push(db)
+      }
+    }
+  }
+
+  getPrimary(): PrimaryDatabase {
+    return this.primary
+  }
+
+  getReplica(tag?: string): Database {
+    if (tag && this.tagged[tag]) {
```

Comment:
We might consider issuing a `log.warn()` the first time the app tries to use a tag that doesn't have a replica.